src/main/java/com/shipengine/InternalClient.java

```diff
@@ -436,44 +443,46 @@ private void checkResponseForErrors(
         switch (statusCode) {
             case 400:
             case 500:
-                Map<String, ArrayList<Map<String, String>>> responseBody400And500 = apiResponseToMap(httpResponseBody);
-                Map<String, String> error400And500 = responseBody400And500.get("errors").get(0);
+                ErrorResponse responseBody400And500 = apiResponseToErrorResponse(httpResponseBody);
+                System.out.println(httpResponseBody);
+                Map<String, String> error400And500 = responseBody400And500.getErrors().get(0);
                 throw new ShipEngineException(
                         error400And500.get("message"),
-                        responseBody400And500.get("request_id").toString(),
+                        responseBody400And500.getRequestId(),
                         error400And500.get("error_source"),
                         error400And500.get("error_type"),
                         error400And500.get("error_code")
                 );
             case 404:
-                Map<String, ArrayList<Map<String, String>>> responseBody404 = httpResponseBody.equals("") ?
-                        Map.of() :
-                        apiResponseToMap(httpResponseBody);
-                Map<String, String> error404 = responseBody404.containsKey("errors") ?
-                        responseBody404.get("errors").get(0) :
+                System.out.println(httpResponseBody);
```

Comment:
@sushithegreat did you mean to leave this `println` here?

Reply:
I did not, thank you 😓